libcontainer/nsenter/nsenter_go122.go

```diff
@@ -0,0 +1,59 @@
+//go:build linux && !gccgo && go1.22
+// +build linux,!gccgo,go1.22
+
+package nsenter
+
+/*
+#cgo LDFLAGS: -Wl,--wrap=x_cgo_getstackbound
+#define _GNU_SOURCE
+#include <pthread.h>
+#include <stdint.h>
+
+typedef uintptr_t uintptr;
+
+// Since Go 1.22 <https://github.com/golang/go/commit/52e17c2>, the
+// go runtime will try to call pthread_getattr_np(pthread_self()) in cgo.
+// This will cause issues in nsexec with glibc < 2.32. so it requires
+// us to provide a backward compatibility with old versions of glibc.
+// The core issue of glibc is that, before 2.32, `pthread_getattr_np`
+// did not call `__pthread_attr_init (attr)`, we need to init the attr
+// in go runtime. Fortunately, cgo exports a function named `x_cgo_getstackbound`,
+// we can wrap it in the compile phrase. Please see:
+// https://github.com/golang/go/blob/52e17c2/src/runtime/cgo/gcc_stack_unix.c
+void __wrap_x_cgo_getstackbound(uintptr bounds[2])
+{
+	pthread_attr_t attr;
+	void *addr;
+	size_t size;
+	int err;
+
+#if defined(__GLIBC__) || (defined(__sun) && !defined(__illumos__))
+	// pthread_getattr_np is a GNU extension supported in glibc.
+	// Solaris is not glibc but does support pthread_getattr_np
+	// (and the fallback doesn't work...). Illumos does not.
+	err = pthread_getattr_np(pthread_self(), &attr);  // GNU extension
+	if (err != 0) {
+		// This is to have a backward compatibility with glibc < 2.32
+		pthread_attr_init(&attr);
+		pthread_getattr_np(pthread_self(), &attr);
+	}
+
+	pthread_attr_getstack(&attr, &addr, &size); // low address
+#elif defined(__illumos__)
```

Comment:
As I said earlier, it makes sense to remove solaris/illumos support, for clarity.

Reply:
I think if the change is small, I want to keep it as is, because we just have only added/changed about 5 lines code.
Then people will read the code more clearly.

Reply:
Makes sense